sdk/python/feast/client.py

```diff
@@ -977,197 +601,128 @@ def ingest(
-            brokers = feature_set.get_kafka_source_brokers()
-            topic = feature_set.get_kafka_source_topic()
-            producer = get_producer(brokers, row_count, disable_progress_bar)
-
-            # Loop optimization declarations
-            produce = producer.produce
-            flush = producer.flush
-            ingestion_id = _generate_ingestion_id(feature_set)
-
-            # Transform and push data to Kafka
-            if feature_set.source.source_type == "Kafka":
-                for chunk in get_feature_row_chunks(
-                    file=dest_path,
-                    row_groups=list(range(pq_file.num_row_groups)),
-                    fs=feature_set,
-                    ingestion_id=ingestion_id,
-                    max_workers=max_workers,
-                ):
-
-                    # Push FeatureRow one chunk at a time to kafka
-                    for serialized_row in chunk:
-                        produce(topic=topic, value=serialized_row)
-
-                    # Force a flush after each chunk
-                    flush(timeout=timeout)
-
-                    # Remove chunk from memory
-                    del chunk
-
-            else:
-                raise Exception(
-                    f"Could not determine source type for feature set "
-                    f'"{feature_set.name}" with source type '
-                    f'"{feature_set.source.source_type}"'
-                )
-
-            # Print ingestion statistics
-            producer.print_results()
-        finally:
-            # Remove parquet file(s) that were created earlier
-            print("Removing temporary file(s)...")
-            shutil.rmtree(dir_path)
-
-        return ingestion_id
+        # Check 1) Only parquet file format for FeatureTable batch source is supported
+        if (
+            feature_table.batch_source
+            and SourceType(feature_table.batch_source.type).name == "BATCH_FILE"
+            and "".join(
+                feature_table.batch_source.file_options.file_format.split()
+            ).lower()
+            != "parquet"
+        ):
+            raise Exception(
+                f"No suitable batch source found for FeatureTable, {name}."
+                f"Only BATCH_FILE source with parquet format is supported for batch ingestion."
+            )
 
-    def get_statistics(
-        self,
-        feature_set_id: str,
-        store: str,
-        features: List[str] = [],
-        ingestion_ids: Optional[List[str]] = None,
-        start_date: Optional[datetime.datetime] = None,
-        end_date: Optional[datetime.datetime] = None,
-        force_refresh: bool = False,
-        project: Optional[str] = None,
-    ) -> statistics_pb2.DatasetFeatureStatisticsList:
-        """
-        Retrieves the feature featureStatistics computed over the data in the batch
-        stores.
+        # Check 2) Check if FeatureTable batch source field mappings can be found in provided source table
```

Comment:
So does this support not providing field mappings, or only providing some field mappings?

Reply:
Not at the moment, it only succeeds if all field mappings are provided in the dataset.